frontend/js/tests/user/Dashboard.test.tsx

```diff
@@ -50,773 +48,763 @@ const props: ListensProps = {
 
-const getComponent = (componentProps: ListensProps) => (
-  <BrowserRouter>
-    <Listens />
-  </BrowserRouter>
+const reactQueryWrapper = ({ children }: any) => (
+  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
 );
 
-// eslint-disable-next-line jest/no-disabled-tests
-xdescribe("Listens page", () => {
-  jest.setTimeout(10000);
+const currentUser = { id: 1, name: "iliekcomputers", auth_token: "fnord" };
 
-  let userEventSession: UserEvent;
+let mockSearchParam = {};
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useSearchParams: () => {
+    const [params, setParams] = React.useState(
+      new URLSearchParams(mockSearchParam)
+    );
+    return [
+      params,
+      (newParams: typeof mockSearchParam) => {
+        mockSearchParam = newParams;
+        setParams(new URLSearchParams(newParams));
+      },
+    ];
+  },
+}));
+
+describe("Listens page", () => {
+  jest.setTimeout(10000);
+  let server: SetupServerApi;
   beforeAll(async () => {
-    userEventSession = await userEvent.setup();
-    fetchMock.enableMocks();
-    fetchMock.mockIf(
-      (input) => input.url.endsWith("/listen-count"),
-      () => {
-        return Promise.resolve(JSON.stringify({ payload: { count: 42 } }));
+    const handlers = [
+      http.post("/", async (path) => HttpResponse.json(props)),
+      http.get("/1/user/*/following", async (path) =>
+        HttpResponse.json({ following: [] })
+      ),
+      http.get("/1/user/*/followers", async (path) =>
+        HttpResponse.json({ followers: [] })
+      ),
+      http.get("/1/user/iliekcomputers/similar-users", async (path) => {
+        HttpResponse.json({ payload: [] });
+      }),
+      http.get("/1/user/*/listen-count", async (path) =>
+        HttpResponse.json({ payload: { count: 42 } })
+      ),
+      http.post("/1/delete-listen", async (path) => HttpResponse.json({})),
+      http.get("/1/user/*/similar-to/*", async (path) =>
+        HttpResponse.json({ payload: [] })
+      ),
+      http.get("/1/stats/user/*/artists", async (path) =>
+        HttpResponse.json({ payload: [] })
+      ),
+    ];
+    server = setupServer(...handlers);
+    server.listen();
+  });
+  afterEach(() => {
+    queryClient.cancelQueries();
+    queryClient.clear();
+  });
+  afterAll(() => {
+    server.close();
+  });
+
+  it("renders correctly on the profile page", async () => {
+    renderWithProviders(
+      <Listens />,
+      {
+        currentUser,
+      },
+      {
+        wrapper: reactQueryWrapper,
+      }
+    );
+
+    await waitFor(() => {
+      // Wait for data to be successfully loaded
+      const state = queryClient.getQueryState(queryKey);
+      expect(state?.status === "success").toBeTruthy();
+    });
+
+    expect(screen.getAllByTestId("listen")).toHaveLength(26);
+  });
+
+  // eslint-disable-next-line jest/no-disabled-tests
+  xit("fetches the user's listen count", async () => {
```

Comment:
This test is still x-ed out and skipped. Is this on purpose?
Probably best not to disable eslint in these cases.

EDIT: I un-x-ed the test and it passes, so I do think this in unintentional.

Reply:
I've had trouble with this test, it passes when I just run a single test, but when I run the whole test suite, it fails. I'm not able to debug this issue

Reply:
Odd. It was passing for me when I ran the Dashboard test suite. I'll check again.